### ISSUE TITLE
Crash in the Android camera-calibration sample

### ISSUE THREAD

**Issue by AleksandrPanov** [opencv/opencv#24612]
### System Information

The latest version OpenCV [version](1ef8175c494dbf9e11fee5913272a2bcc2f6002f) was used

devices:

- Huawei P20 Pro, EMUI 12.0.0, (Android 10?)
- Xiaomi Mi 8, Android 10, MIUI 12.0.3

### Detailed description

The **problem may be related** to https://github.com/android/ndk/issues/289

Problem in findCirclesGrid() call, this func throw error:
```
2023-11-28 20:06:07.539  9151-9233  cv::error()             org...ncv.samples.cameracalibration  E  OpenCV(4.8.0-dev) Error: No Error (samples is empty) in filterOutliersByDensity, file ../../modules/calib3d/src/circlesgrid.cpp, line 1098
2023-11-28 20:06:07.541  9151-9233  libc                    org...ncv.samples.cameracalibration  A  Fatal signal 11 (SIGSEGV), code 2
```
Problem in this code:
```
if (samples.empty())
    CV_Error( 0, "samples is empty" );
```

This `CV_Error( 0, "samples is empty" );` is located in `try/catch` block, but in this Android sample and my device this causes crash anyway.

The error is not reproduced on all devices.

[error_logs.txt](https://github.com/opencv/opencv/files/13491420/error_logs.txt)

### Steps to reproduce
Build with build_sdk.py
```
export ANDROID_NDK_HOME=/Android/Sdk/ndk/18.1.5063045/
export ANDROID_SDK=/Android/Sdk/
python /opencv/platforms/android/build_sdk.py /opencv/build/ /home/alexander/all/opencv/opencv --ndk_path /Android/Sdk/ndk/18.1.5063045/ --sdk_path /Android/Sdk/ --config /opencv/platforms/android/ndk-18-api-level-21.config.py
```
run camera-calibration sample

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by AleksandrPanov** [comment]
The problem is solved by using a newer version of NDK (NDK 25.2.9519653 was tested).

**Comment by AleksandrPanov** [comment]
Use a newer version of NDK (25.2.9519653+) to avoid problems.
